src/peft/tuners/loha/layer.py

```diff
@@ -127,14 +135,35 @@ def update_layer(
         if isinstance(base_layer, nn.Linear):
             shape = tuple(base_layer.weight.shape)
         elif isinstance(base_layer, nn.Conv2d):
-            use_effective_conv2d = use_effective_conv2d and base_layer.kernel_size != (1, 1)
+            # Handle 1x1 convolutions differently
+            if base_layer.kernel_size == (1, 1):
+                # For 1x1 convolutions, use a more direct shape without using effective_conv2d
+                shape = (base_layer.out_channels, base_layer.in_channels, *base_layer.kernel_size)
+                use_effective_conv2d = False
+            else:
+                use_effective_conv2d = use_effective_conv2d
             if use_effective_conv2d:
                 shape = (base_layer.out_channels, base_layer.in_channels, *base_layer.kernel_size)
             else:
                 shape = (
                     base_layer.out_channels,
                     base_layer.in_channels * base_layer.kernel_size[0] * base_layer.kernel_size[1],
                 )
+        elif isinstance(base_layer, nn.Conv1d):
+            # Handle kernel_size=1 Conv1d differently
+            if base_layer.kernel_size[0] == 1:
+                # For kernel_size=1, use a more direct shape without using effective_conv2d
+                shape = (base_layer.out_channels, base_layer.in_channels, base_layer.kernel_size[0])
+                use_effective_conv2d = False
```

Comment:
This shape is never used as it is overridden by the second condition (`else`-branch).